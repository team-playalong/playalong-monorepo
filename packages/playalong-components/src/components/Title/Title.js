import * as React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';

const TitleComp = styled.h1`
  font-size: 20px;
  font-weight: 500;
  letter-spacing: .005em;
  font-family: Roboto;
`;

Title.propTypes = {
  text: string
};
function Title ({ text = '' }) {
  return <TitleComp>{text}</TitleComp>;
}

export const props = ['text'];
export default Title;
