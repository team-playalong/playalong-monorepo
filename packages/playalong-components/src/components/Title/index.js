export {default, props} from './Title';
