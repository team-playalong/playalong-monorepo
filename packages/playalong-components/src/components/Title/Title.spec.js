import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Title from './Title';

let wrapper;
const props = {
  text: 'Hello!',
};

beforeEach(() => {
  wrapper = shallow(<Title {...props} />);
});

test('Title should render text', () => {
  expect(wrapper.html()).toContain(props.text);
});

test('Snapshot testing', () => {
  const tree = renderer.create(<Title {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
