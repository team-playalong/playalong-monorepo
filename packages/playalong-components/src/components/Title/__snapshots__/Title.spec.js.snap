// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<h1
  className="sc-bdVaJa gHcgxb"
>
  Hello!
</h1>
`;
